packages/core/react/src/useAnchorWallet.ts

```diff
@@ -1,11 +1,11 @@
-import type { PublicKey, Transaction } from '@solana/web3.js';
+import type { PublicKey, Transaction, VersionedTransaction } from '@solana/web3.js';
 import { useMemo } from 'react';
 import { useWallet } from './useWallet.js';
 
 export interface AnchorWallet {
     publicKey: PublicKey;
-    signTransaction(transaction: Transaction): Promise<Transaction>;
-    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signTransaction(transaction: Transaction | VersionedTransaction): Promise<Transaction | VersionedTransaction>;
+    signAllTransactions(transactions: (Transaction | VersionedTransaction)[]): Promise<(Transaction | VersionedTransaction)[]>;
```

Comment:
This will likely need to use generics -- the type of the transactions returned should match the type of the inputs.

Reply:
I've updated this, but not sure if it is what you meant. I'm not very familiar with using generics in typescript.

Reply:
You did it right!